supabase/functions/vapi-assistant-provision/index.ts

```
@@ -231,10 +286,33 @@ Deno.serve(async (req: Request) => {
         user_id,
       });
     }
-    if (!displayName) displayName = email;
+    // Sanitize displayName before using it in prompts/logs: single-line, bounded
+    {
+      const baseName =
+        typeof displayName === 'string' && displayName.trim().length > 0
+          ? displayName
+          : email;
+      // Replace control characters (including newlines/tabs) with spaces
+      const noCtrl = String(baseName).replace(/[\x00-\x1F\x7F]+/g, ' ');
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)